src/components/Tracker/Activity.js

```diff
@@ -0,0 +1,471 @@
+        <VictoryAxis
+            tickCount={props.data.length}
+            tickFormat={date => (new Date(date)).getDate()}
+            style={sharedAxisStyles}
+        />
+        <VictoryAxis
+            dependentAxis
+            label="Distance in miles"
+            style={sharedAxisStyles}
+        />
+        <VictoryLegend x={270} y={60}
+            standalone={false}
+            orientation="horizontal"
+            gutter={20}
+            style={{ border: { stroke: "black" } }}
+            colorScale={[navy, red, orange]}
+            data={[
+                { name: "Walked" }, { name: "Ran" }, { name: "Cycled" }
+            ]}
+        />
+    </VictoryChart>
+);
+
+const MinutesMonthly = (props) => (
+    <VictoryChart scale={{ x: "time" }}
+        padding={{ top: 50, bottom: 50, left: 100, right: 10 }}
+        width={1000} height={450}
+    >
+        <VictoryLabel
+            x={300}
+            y={10}
+            style={[{ fontSize: 30 }]}
+            textAnchor="middle"
+            text="Minutes in the last 30 days"
+        />
+        <VictoryArea
+            style={{
+                data: { fill: navy },
+            }}
+            data={props.data}
+            x="date"
+            y="minutes"
+        />
+        <VictoryAxis
+            tickCount={props.data.length}
+            tickFormat={date => (new Date(date)).getDate()}
+            style={sharedAxisStyles}
+        />
+        <VictoryAxis
+            dependentAxis
+            label="Minutes Exercised"
+            style={sharedAxisStyles}
+        />
+    </VictoryChart>
+);
+
+export default class Activity extends Component {
+    constructor(props) {
+        super(props);
+        this.getHealthData = this.getHealthData.bind(this);
+
+        this.state = {
+            todaysData: [],
+            weeklyData: [],
+            monthlyData: [],
+        }
+    }
+
+    componentDidMount() {
+        if (this.props.user.token) this.getHealthData();
+    }
+
+    async getHealthData() {
+        try {
+            const dailyData = await getDailyHealth(this.props.user.token);
+            const weekData = await getWeeklyHealth(this.props.user.token);
+            const monthData = await getMonthlyHealth(this.props.user.token);
+            console.log(weekData);
+            console.log(monthData);
+
+            this.setState({ todaysData: dailyData, weeklyData: weekData, monthlyData: monthData });
+        } catch (err) {
+            console.error(err.message);
+        }
+    }
+
+    // will need to create a flexbox to align the charts
+    render() {
+        return (
+            <>
+                <Navigation {...this.props} />
+                <div className="row content">
+                    <div className="container">
+                        <div className="pie">
+                            <CaloriesDaily
+                                burned={this.state.todaysData.length > 0 ? this.state.todaysData[0].calories : 0}
```

Comment:
the activity view was not updating when i tried inputting data more than once and i think it's bc we're using index 0 each time to render. if we change it to `this.state.todaysData[this.state.todaysData.length - 1]` this seems to get the most recent data

Reply:
You shouldn't be inputting daily data more than once, the assumption is that the user is only inputting data once a day. But that fix could work in case they do input data more than once a day

Reply:
ohhh so would the daily data shown on activity change based on the date?

Reply:
yeah! it fetches all the data based on today's date